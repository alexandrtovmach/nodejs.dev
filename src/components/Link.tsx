import { Link } from 'gatsby';
import React from 'react';

const UniversalLink = props => {
  const { to, children } = props;
  if (to.includes('http')) {
    // render absolute link
    return (
      <a href={to} target="_blank" rel="noopener noreferrer" {...props}>
        {children}
      </a>
    );
  }
  // render gatsby optimized link
  return (
    <Link to={to} {...props}>
      {children}
    </Link>
  );
};

export default UniversalLink;
