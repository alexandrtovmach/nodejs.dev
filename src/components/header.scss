.nav {
  background: var(--gray0);
  box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.33), 0 2px 3px rgba(0, 0, 0, 0.2);
  height: var(--nav-height);
  line-height: var(--nav-height);
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 999;
}

.ui-styles {
	@media (max-width: 380px) {
    padding: 0;
  }
  margin: 0 auto;
  padding: 0 4.8rem;
  display: flex;
  align-items: center;
  list-style: none;
}