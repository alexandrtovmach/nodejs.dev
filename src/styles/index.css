

.pentagon-illustration-big1 {
    position: absolute;
    top: 1200px;
    z-index: -1;
    right: 0;
}

.pentagon-illustration-big2 {
    position: absolute;
    z-index: -2;
    left: 0;
    bottom: -2000px;
}

.double-background {
    position: absolute;
    width: 100%;
    height: 1954px;
    top: 808px;
    left: 0;
    background: var(--color-fill-canvas);
    z-index: -10;
}

.btn-subscribe {
    border: none;
    box-sizing: border-box;
    height: 5.6rem;
    padding: 0 var(--space-24);
    font-weight: var(--font-weight-semibold);
    background: var(--color-fill-action);
    border-radius: var(--space-04);
    color: var(--black0);
    margin: 0;
    margin-left: var(--space-08);
    cursor: pointer;
}

.trusted-by {
    margin-bottom: var(--section-margin-bottom);
    text-align: center;
    width: 100%;
}

.trusted-by p {
    color: var(--color-text-secondary);
    margin-bottom: var(--space-64);
}

.trusted-by .logos-container {
    display: flex;
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
    filter: grayscale();
}

.get-started-callouts {
    width: 100%;
    display: flex;
    flex-direction: row;
    text-align: center;
    justify-content: space-between;
    margin-bottom: var(--space-80);
}

.get-started-callout {
    text-decoration: none;
    display: flex;
    flex-direction: column;
}

.get-started-callout h5 {
    color: var(--color-text-primary);
    margin: 0;
    margin-top: var(--space-24);
}

.get-started-callout p {
    margin-top: var(--space-16);
    color: var(--color-text-secondary);
    width: 422px;
}

.btn-primary {
    display: block;
    width: 141px;
    margin: 0 auto;
    cursor: pointer;
    border: none;
    box-sizing: border-box;
    padding: var(--space-08) var(--space-24);
    font-weight: var(--font-weight-semibold);
    background: var(--color-fill-action);
    color: var(--black0);
    border-radius: var(--space-04);
    text-decoration: none;
    line-height: var(--line-height-body1);
}

.btn-primary:hover, .btn-primary:focus {
    color: var(--black0);
}
  