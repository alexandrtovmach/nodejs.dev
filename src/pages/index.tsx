import React from 'react';

import MainSection from '../components/LandingPage/Sections/MainSection/main-section';
import SandboxSection from '../components/LandingPage/Sections/SandboxSection/sandbox-section';
import Layout from '../templates/Layout/layout';
import Link from '../components/Link/Link';

import '../styles/index.css';

import featureImg from '../images/placeholder-img.png';

import logoImg1 from '../images/logos/ibm-logo.svg';
import logoImg2 from '../images/logos/linkedin-logo.svg';
import logoImg3 from '../images/logos/microsoft-logo.svg';
import logoImg4 from '../images/logos/netflix-logo.svg';
import logoImg5 from '../images/logos/paypal-logo.svg';

import GetStartedIllustration1 from '../images/illustrations/beginners-guide-illustration.svg';
import GetStartedIllustration2 from '../images/illustrations/do-more-illustration.svg';

import pentagonIllustration1 from '../images/illustrations/pentagon-illustration1.svg';
import pentagonIllustration2 from '../images/illustrations/pentagon-illustration2.svg';

// should be reaplaced with GraphQL data
const title = 'Node.js Official website';
const description = 'You have hit a route that does not exist.';
const content = {
  section1_1: 'The power of JavaScript minus the browser',
  section1_2: `Tacos raw denim palo santo, squid 90's occupy sustainable leggings locavore before they sold out chambray gastropub synth. Cornhole lomo dreamcatcher celiac enamel pin, yr lyft master cleanse meggings.`,
};

const nodeFeature1 =
  'Lorem ipsum dolor amet pug vape +1 poke pour-over kitsch tacos meh. ';
const nodeFeature2 =
  'Lorem ipsum dolor amet pug vape +1 poke pour-over kitsch tacos meh. ';
const nodeFeature3 =
  'Lorem ipsum dolor amet pug vape +1 poke pour-over kitsch tacos meh. ';

const NodeFeature = ({ img, featureText }: Props) => {
  return (
    <div className="node-features__feature">
      <img src={img} alt="node feature" />
      <p className="t-caption">{featureText}</p>
    </div>
  );
};

interface Props {
  img: string;
  featureText: string;
}

export default function Index(): JSX.Element {
  const description = 'Welcome to Node.js!';

  return (
    <Layout title={title} description={description}>
      <MainSection content={content} />
      <SandboxSection content={content} />

      <section className="node-features">
        <NodeFeature img={featureImg} featureText={nodeFeature1} />
        <NodeFeature img={featureImg} featureText={nodeFeature2} />
        <NodeFeature img={featureImg} featureText={nodeFeature3} />
      </section>

      <section className="join-node">
        <h4 className="t-headline">
          <span className="accent">Join</span> the community
        </h4>
        <div className="join-node-form-container">
          <p className="t-body2">
            We’ll never share your information and always respect your inbox -
            quality content only, we promise.{' '}
          </p>
          <div style={{ display: `flex`, alignItems: 'flex-start' }}>
            <input
              type="email"
              placeholder="node@nodejs.dev"
              className="input-subscribe"
            />
            <button className="btn-subscribe t-body1" type="button">
              Subscribe
            </button>
          </div>
        </div>
      </section>

      <section className="trusted-by">
        <h4 className="t-headline">
          Trusted by development teams around the world
        </h4>
        <p>Including IBM, LinkedIn, Microsoft, Netflix, and PayPal.</p>
        <div className="logos-container">
          <img src={logoImg1} alt="ibm logo" />
          <img src={logoImg2} alt="linkedin logo" />
          <img src={logoImg3} alt="microsoft logo" />
          <img src={logoImg4} alt="netflix logo" />
          <img src={logoImg5} alt="paypal logo" />
        </div>
      </section>

      <section className="get-started-callouts">
        <Link to="/learn" className="get-started-callout">
          <img src={GetStartedIllustration1} alt="" />
          <h5 className="t-headline">A beginner’s guide</h5>
          <p>
            Lorem ipsum dolor amet pug vape +1 poke pour-over kitsch tacos meh.
          </p>
        </Link>

        <Link to="/docs" className="get-started-callout">
          <img src={GetStartedIllustration2} alt="" />
          <h5 className="t-headline">Do even more with Node</h5>
          <p>
            Lorem ipsum dolor amet pug vape +1 poke pour-over kitsch tacos meh.
          </p>
        </Link>
      </section>

      <Link to="/learn" className="btn-primary">
        Get Started
      </Link>
      <img
        className="pentagon-illustration-big1"
        src={pentagonIllustration1}
        alt=""
      />
      <img
        className="pentagon-illustration-big2"
        src={pentagonIllustration2}
        alt=""
      />
      <div className="double-background" />
    </Layout>
  );
}
