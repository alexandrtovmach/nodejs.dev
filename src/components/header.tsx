import { Link } from 'gatsby';
import React from 'react';
import logo from '../images/logo.svg';
import './header.scss';


const Header = () => (
	<header className="main-header">
		<nav className="nav">
			<ul className="ui-styles">
				<li>
					<Link to="/" style={{ display: 'block' }}>
						<img src={logo} alt="Node.js" className="nav__logo" />
					</Link>
				</li>
				<li className="nav__tabs">
					<a target="_blank" href="https://nodejs.org/en/download/">
						Download
					</a>
				</li>
				<li className="nav__tabs">
					<a target="_blank" href="https://nodejs.org/en/docs/">
						API Docs
					</a>
				</li>
				<li className="nav__tabs">
					<a target="_blank" href="https://github.com/nodejs/nodejs.dev">
						GitHub
					</a>
				</li>
			</ul>
		</nav>
	</header>
);

export default Header;
