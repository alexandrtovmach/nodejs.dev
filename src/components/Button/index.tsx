import React from 'react';
import { Link } from '..';

import styles from './button.module.scss';

interface ButtonProps {
  children: JSX.Element[] | JSX.Element | string;
  type?: 'primary' | 'secondary' | 'default';
  shape?: 'rounded';
  disabled?: boolean;
  to?: string;
  alt?: string;
  onClick?: () => {};
  className?: string;
}

const Button = (props: ButtonProps): JSX.Element => {
  const {
    children,
    type,
    disabled,
    shape,
    to,
    onClick,
    className,
    alt,
  } = props;
  if (to && !onClick) {
    return (
      <Link
        to={to}
        className={[
          styles.button,
          styles[disabled ? 'disabled' : type || 'primary'],
          shape && styles[shape],
          className,
        ].join(' ')}
        disabled={disabled}
        title={alt}
      >
        {children}
      </Link>
    );
  }
  return (
    <button
      className={[
        styles.button,
        styles[disabled ? 'disabled' : type || 'primary'],
        shape && styles[shape],
        className,
      ].join(' ')}
      disabled={disabled}
      onClick={onClick}
      title={alt}
      type="button"
    >
      {children}
    </button>
  );
};

export default Button;
