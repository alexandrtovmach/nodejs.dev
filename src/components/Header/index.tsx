import React from 'react';

import { Link } from '..';
import DarkModeController from '../../util/DarkModeController';
import logoLight from '../../images/logo-light.svg';
import logoDark from '../../images/logo-dark.svg';
import styles from './header.module.scss';

// should be reaplaced with GraphQL data

const content = {
  navigationLinks: [
    {
      link: 'learn',
      text: 'Learn',
    },
    {
      link: 'docs',
      text: 'Documentation',
    },
    {
      link: 'download',
      text: 'Download',
    },
  ],
  toggleTheme: 'Toggle theme',
};

interface HeaderProps {
  darkModeController?: DarkModeController;
}

const Header = ({ darkModeController }: HeaderProps): JSX.Element => (
  <header className={styles.mainHeader}>
    <nav className={styles.navigation}>
      <Link to="/" noActiveState>
        <img
          src={logoLight}
          alt="Node.js"
          className={[styles.navigationLogo, 'light-mode-only'].join(' ')}
        />
        <img
          src={logoDark}
          alt="Node.js"
          className={[styles.navigationLogo, 'dark-mode-only'].join(' ')}
        />
      </Link>
      {content.navigationLinks.map(
        (item, i): JSX.Element => (
          <Link
            to={item.link}
            key={`header-navigation-link-${i * 3.14}`}
            className={styles.navigationItem}
          >
            {item.text}
          </Link>
        )
      )}
    </nav>
    <div className={styles.panel}>
      <div className={styles.navigationItem}>
        <button
          className={styles.themeToggler}
          onClick={() =>
            !darkModeController && document.body.classList.toggle('dark-mode')
          }
          onPointerDown={(): void =>
            darkModeController && darkModeController.onPointerDown()
          }
          onPointerUp={(): void =>
            darkModeController && darkModeController.onPointerUp()
          }
          title={content.toggleTheme}
          type="button"
        >
          <i className="material-icons light-mode-only">nights_stay</i>
          <i className="material-icons dark-mode-only">wb_sunny</i>
        </button>
      </div>
    </div>
  </header>
);

export default Header;
