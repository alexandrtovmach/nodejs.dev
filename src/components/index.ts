import Link from './Link/link';
import Button from './Button/button';
import Header from './Header/header';

import CommunitySection from './LandingPage/Sections/CommunitySection/community-section';
import FeaturesSection from './LandingPage/Sections/FeaturesSection/features-section';
import GetStartedSection from './LandingPage/Sections/GetStartedSection/getstarted-section';
import MainSection from './LandingPage/Sections/MainSection/main-section';
import PartnerSection from './LandingPage/Sections/PartnerSection/partner-section';
import SandboxSection from './LandingPage/Sections/SandboxSection/sandbox-section';

export {
  Link,
  Button,
  Header,
  CommunitySection,
  FeaturesSection,
  GetStartedSection,
  MainSection,
  PartnerSection,
  SandboxSection,
};
