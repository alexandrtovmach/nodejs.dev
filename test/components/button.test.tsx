import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../../src/components';

describe('Tests for Button component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button>Custom</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
