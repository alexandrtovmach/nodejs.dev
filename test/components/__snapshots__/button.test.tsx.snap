// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Button component renders correctly 1`] = `
<button
  className="button primary   "
  type="button"
>
  Custom
</button>
`;
